lib/vector-core/src/event/metric.rs

```diff
@@ -593,7 +593,7 @@ impl MetricValue {
                 *samples = samples
                     .iter()
                     .copied()
-                    .filter(|sample| samples2.iter().find(|sample2| sample == *sample2).is_none())
+                    .filter(|sample| !samples2.iter().any(|sample2| sample == sample2))
```

Comment:
Not this?
```suggestion
                    .filter(|sample| samples2.iter().all(|sample2| sample != sample2))
```

Reply:
Hmm, worth a follow-up PR. I wonder if `all` is able to bail out faster than `any`. This form, anyway, was suggested by clippy.

Reply:
I doubt either is faster, as it's the same steps needed, just a matter of where the negations go in order to express the intent clearer.